### PR TITLE
RATEST-232: Failure when validating the vital value of respiratory rate

### PR DESCRIPTION
**Pull request by kdaud**
Ticket [ID](https://issues.openmrs.org/browse/RATEST-232)

Description -> Failure when validating the vital value of respiratory rate

### CODE REVIEW

**Comment by kdaud** [comment]
Waiting for review cc: @sherrif10

**Comment by sherrif10** [comment]
Hello @kdaud ,Resolve the build failures then this looks good

**Comment by dkayiwa** [comment]
@kdaud have the github actions instance run the same versions of modules as qa-refapp

**Comment by kdaud** [comment]
> @kdaud have the github actions instance run the same versions of modules as qa-refapp

@dkayiwa I haven’t studied this in too much detail, But I think they have different version based on the logs produced by cumber report at https://reports.cucumber.io/reports/cdbfa2cd-a0a1-4231-96e5-c79c9a08aba4e. 

Am thinking making the local instance have the same versions of modules as qa-refapp is the smart way to go. @sherrif10 do you have an idea on how to achieve the goal?

**Comment by kdaud** [comment]
The build failure is resolved and waiting for more recommendation if any  cc: @sherrif10

**Comment by sherrif10** [comment]
Is this ticket supposed to resolve failure on this https://ci.openmrs.org/browse/CONTRIB-QA-815

File: src/test/java/org/openmrs/contrib/qaframework/automation/VitalsAndTriagingSteps.java [hunk 1]
```diff
@@ -118,11 +118,11 @@ public void crossCheckMaximumVitalsValidity() {
 				hasItem("Maximum: 230"));
 		patientCaptureVitalsPage.clearPatientPulse();
 		patientCaptureVitalsPage.setPulseField("78");
-		patientCaptureVitalsPage.setRespiratoryField("1004");
+		patientCaptureVitalsPage.setRespiratoryField("106");
```
Comment (sherrif10) [inline src/test/java/org/openmrs/contrib/qaframework/automation/VitalsAndTriagingSteps.java:121]: This repository field is the caus of the failures. can you please change this to maximum value less than 99. this implies you are still testing on demo server which is not up to date with a server.

**Comment by kdaud** [comment]
> Is this ticket supposed to resolve failure on this https://ci.openmrs.org/browse/CONTRIB-QA-815

Yes

**Comment by sherrif10** [comment]
> Yes

It will fail still if you are testing this functionality on demo server, this change is not yet released in reference demo data= demo server. you need to enter value less than 99 in respiratory field

**Comment by sherrif10** [comment]
Try to test this functionality on demo server and qa instance to have the difference of values to be entered

File: src/test/java/org/openmrs/contrib/qaframework/automation/VitalsAndTriagingSteps.java (hunk 1 above)
Comment (kdaud) [reply src/test/java/org/openmrs/contrib/qaframework/automation/VitalsAndTriagingSteps.java:121]: > can you please change this to maximum value less than 99

The method [crossCheckMaximumVitalsValidity()](https://github.com/openmrs/openmrs-contrib-qaframework/blob/dc7ff801d2126546a8b58eb56449f9986a5c5611/src/test/java/org/openmrs/contrib/qaframework/automation/VitalsAndTriagingSteps.java#L100) is asserting that a vital can be saved in the database if its above the maximum value of that given parameter.  

For this test to validate the respiratory rate value, the first entered value must be above the maximum value `patientCaptureVitalsPage.setRespiratoryField("106");` and then it validates the value from at line code [122](https://github.com/openmrs/openmrs-contrib-qaframework/blob/dc7ff801d2126546a8b58eb56449f9986a5c5611/src/test/java/org/openmrs/contrib/qaframework/automation/VitalsAndTriagingSteps.java#L122) and then the test sets the value that is within the range at code line [125](https://github.com/openmrs/openmrs-contrib-qaframework/blob/dc7ff801d2126546a8b58eb56449f9986a5c5611/src/test/java/org/openmrs/contrib/qaframework/automation/VitalsAndTriagingSteps.java#L125)

**Comment by kdaud** [comment]
> It will fail still if you are testing this functionality on demo server

Bamboo uses qa instance to run builds for [QA Framework](https://ci.openmrs.org/browse/CONTRIB-QA), so I guess Bamboo build plan will be green once this commit enters into master.

**Comment by kdaud** [comment]
> Try to test this functionality on demo server and qa instance to have the difference of values to be entered

Testing the commit changes on demo server fails the build plan due to the validation [here](https://github.com/openmrs/openmrs-contrib-qaframework/blob/dc7ff801d2126546a8b58eb56449f9986a5c5611/src/test/java/org/openmrs/contrib/qaframework/automation/VitalsAndTriagingSteps.java#L122) which is not true with what the server is reading.

On the other hand, testing the changes on qa instance passes the test because the value set at line code [125](https://github.com/openmrs/openmrs-contrib-qaframework/blob/dc7ff801d2126546a8b58eb56449f9986a5c5611/src/test/java/org/openmrs/contrib/qaframework/automation/VitalsAndTriagingSteps.java#L125) for respiratory rate is within the acceptable range of the parameter.

**Comment by kdaud** [comment]
@sherrif10 any more recommendation to address?

**Review by sherrif10 — APPROVED** [review]
Hello @dkayiwa  .This looks cleaner ad ready to be merged. Thanks @kdaud

**Comment by sherrif10** [comment]
Have also faced this fix and fixed it in here https://github.com/openmrs/openmrs-contrib-qaframework/pull/127

File: src/test/java/org/openmrs/contrib/qaframework/automation/VitalsAndTriagingSteps.java [hunk 2]
```diff
@@ -118,11 +118,11 @@ public void crossCheckMaximumVitalsValidity() {
...
 		assertThat(patientCaptureVitalsPage.getValidationErrors(),
-				hasItem("Maximum: 999"));
+				hasItem("Maximum: 99"));
 		patientCaptureVitalsPage.clearPatientRespiratoryRate();
-		patientCaptureVitalsPage.setRespiratoryField("14");
```
Comment (dkayiwa) [inline src/test/java/org/openmrs/contrib/qaframework/automation/VitalsAndTriagingSteps.java:125]: Why the change from 14 to 16?

File: src/test/java/org/openmrs/contrib/qaframework/automation/VitalsAndTriagingSteps.java [hunk 3]
```diff
@@ -118,11 +118,11 @@ public void crossCheckMaximumVitalsValidity() {
 				hasItem("Maximum: 230"));
 		patientCaptureVitalsPage.clearPatientPulse();
 		patientCaptureVitalsPage.setPulseField("78");
-		patientCaptureVitalsPage.setRespiratoryField("1004");
```
Comment (dkayiwa) [inline src/test/java/org/openmrs/contrib/qaframework/automation/VitalsAndTriagingSteps.java:121]: Can you also explain the change from 1004 to 106?

**Comment by kdaud** [comment]
I imagined myself as a data clerk entering patient vitals and instead of entering 16 for respiratory rate, I unfortunately enter 106.

> Why the change from 14 to 16?

> Can you also explain the change from 1004 to 106?

The test was written when the maximum value the system could record for respiratory rate was 999 and the think fully the developer decided to use 1004 and 14 for the test when validating the vital.

Now that the maximum value of respiratory rate that can be recorded is 99, I opted to change to a value slight above the maximum. 1004 is far above 99 than 106, 106 overrides 1004.

Why I opted to use 106 and 16?
Usually reliable data should not have a big value of the absolute error. Using 1004 and 14 has a bigger absolute error value compared to 106 and 16. 

Alternatively we can use 104 & 14 for the test than using 1004 & 14. This would mean changing 1004 to 104

**Comment by dkayiwa** [comment]
@kdaud are those changes necessary to fix the reported problem?

**Comment by kdaud** [comment]
> @kdaud are those changes needed to fix the reported problem?

`Boolean areThoseChangesNeededToFixTheReportedProblem = false;`

**Comment by kdaud** [comment]
All the reviews are caught with care, and now waiting for a merge trigger for the changes to enter master!
cc: @dkayiwa
